device-types/Arista/DCS-7358.yaml

```diff
@@ -0,0 +1,49 @@
+---
+manufacturer: Arista
+model: DCS-7358
+slug: arista-dcs-7358
+part_number: DCS-7358
+u_height: 4
+is_full_depth: false
+comments: '[Arista 7358X4 Series 100/200/400G Data Center Switches](https://www.arista.com/assets/data/pdf/Datasheets/7358X4-Datasheet.pdf'
+power-ports:
+  - name: PS1
+    type: iec-60320-c20
+    maximum_draw: 2000
+  - name: PS2
+    type: iec-60320-c20
+    maximum_draw: 2000
+  - name: PS3
+    type: iec-60320-c20
+    maximum_draw: 2000
+  - name: PS4
+    type: iec-60320-c20
+    maximum_draw: 2000
```

Comment:
Are these PSU's modular? If so, then they should be modelled here under `module-bays` and the associated PSU's should be created in the module-types folder if they do not already exist